ICSharpCode.Decompiler/IL/Transforms/TransformArrayInitializers.cs

```diff
@@ -143,6 +143,40 @@ internal static bool TransformSpanTArrayInitialization(NewObj inst, StatementTra
 			return false;
 		}
 
+		internal static bool TransformRuntimeHelpersCreateSpanInitialization(Call inst, StatementTransformContext context, out ILInstruction replacement)
+		{
+			replacement = null;
+			if (!context.Settings.ArrayInitializers)
+				return false;
+			if (MatchRuntimeHelpersCreateSpan(inst, context, out var elementType, out var field))
+			{
+				if (field.HasFlag(System.Reflection.FieldAttributes.HasFieldRVA))
+				{
+					var valuesList = new List<ILInstruction>();
+					var initialValue = field.GetInitialValue(context.PEFile, context.TypeSystem);
+					var elementTypeSize = elementType.GetSize();
+					if (elementTypeSize <= 0 || initialValue.Length % elementTypeSize != 0)
+						return false;
+
+					var size = initialValue.Length / elementTypeSize;
+					if (context.Settings.Utf8StringLiterals &&
+						elementType.IsKnownType(KnownTypeCode.Byte) &&
+						DecodeUTF8String(initialValue, size, out string text))
+					{
+						replacement = new LdStrUtf8(text);
+						return true;
+					}
+					if (DecodeArrayInitializer(elementType, initialValue, new[] { size }, valuesList))
+					{
+						var tempStore = context.Function.RegisterVariable(VariableKind.InitializerTarget, new ArrayType(context.TypeSystem, elementType));
+						replacement = BlockFromInitializer(tempStore, elementType, new[] { size }, valuesList.ToArray());
+						return true;
+					}
```

Comment:
This is the same as for `Span<T>(void*, int)` handler. Should split this out into a shared function?

Reply:
You could do that, yes...